subgraph/src/public-lock.ts

```diff
@@ -314,3 +322,34 @@ export function handleLockMetadata(event: LockMetadataEvent): void {
     lock.save()
   }
 }
+/**
+ * Create Receipt object for subgraph for key 'purchase'/'extend'/'renewal'
+ * @param {String} keyID - key id
+ * @param {event} Object - Object event
+ * @return {void}
+ */
+export async function createReceipt(
+  event: RenewKeyPurchaseEvent | TransferEvent | KeyExtendedEvent
+): Promise<void> {
+  const hash = event.transaction.hash.toString()
+  const receipt = new Receipt(hash)
+  const lock = Lock.load(event.address.toHexString())
+  const key = Key.load(hash)
+  const hasErc20 = (lock?.tokenAddress ?? '')?.length > 0
+
+  if (hasErc20) {
+    // todo
+    console.log(event.receipt?.logs)
```

Comment:
const logs = event.receipt?.logs

if(logs.length) {
// find the ERC20 Transfer event
const log = logs.find(l => l.address = lock.tokenAddress && l.topics[0] === kecakk256("Transfer(from, to, value)"))

// decode event data
let decoded = ethereum.decode('(address,address,uint256)', log.topics)
receipt.payer = decoded[0]
receipt.amount = decoded[2]
}